FEPLT-2075: storybook migration b4

sarkaaa opened the pull request:
# This Pull Request meets the following criteria:

### For new components:

- [ ] Unit and visual regression tests have been added/adjusted for my new feature
- [ ] New Components are registered in index.js of my project
- [ ] New Components have `d.ts` files and are exported in `index.d.ts`


Closes https://kiwicom.atlassian.net/browse/FEPLT-2075

sarkaaa commented on packages/orbit-components/src/NotificationBadge/NotificationBadge.stories.tsx:
```diff
@@ -8,236 +9,86 @@ import NotificationBadge from ".";
-  return <NotificationBadge type={TYPE_OPTIONS.NEUTRAL}>{content}</NotificationBadge>;
-};
 
-Neutral.story = {
```
For this Story file, there were soooo many similar stories (the differences were only in `type` prop), so I squash them into one story with `type` option.

sarkaaa commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,113 @@
-    control: {
-      type: "select",
+  parameters: {
+    info: "Playground of Loading component. Type buttonLoader inherits parent's stroke color. For visualisation of undefined value for customSize arg, set value number 0. Visit Orbit.Kiwi for more detailed guidelines.",
```
Some of the component props depend on the specific configuration, so I wrote some extended instructions and explanations.

sarkaaa commented on packages/orbit-components/src/NotificationBadge/NotificationBadge.stories.tsx:
```diff
@@ -8,238 +9,97 @@ import NotificationBadge from ".";
-    </div>
-  );
-};
+export const Playground: Story = {
```
I was considering squashing the `Accessibility` and `Playground` stories as they're almost the same (Accessibility includes `ariaLabel` attribute. What's your opinion?

domihustinova replied:
I agree. Usually (or maybe even for every component), there is more controls in the Playground story but here Accessibility story has one extra control. If we would keep the Accessibility story, it would maybe make more sense to leave there only one control -  the `ariaLabel` one. But I would prefer just keeping Default, Playground and RTL ones.

sarkaaa commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,113 @@
-InlineLoader.args = {
-  loadingText: "Please wait, content of the page is loading...",
-};
+export const DefaultWithChildren: Story = {
```
What do you suggest to add there instead of the basic text component? I just added this `<Text>` component for simple visualization, but I can add there more (eg. illustration, alert components, whatever,...)

sarkaaa commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,113 @@
-export const Playground = ({ type, loadingText, loading, dataTest }) => {
-  return <Loading loading={loading} type={type} text={loadingText} dataTest={dataTest} />;
-};
+export const Playground: Story = {
```
Card component story - I removed it as I found it redundant because the Card wrapper just wraps `loading` component and the children is the same as `Default with children` story.

`loading` missing in playground - the example in Playground story doesn't have any children and the behaviour is same for `false`/`true` value for `loading` prop.

text in Playground - this prop is `React.ReactNode` type, the changes text string need to be in `"`, either it's not working. I was thinking it can be confusing for the user and having the `text` as a constant ReactNode element in the example code is more clear (?). Idk, what would be better, what's your opinion?

![image](https://github.com/user-attachments/assets/6835dd30-8860-468b-ab13-30f5a03fccd5)

![image](https://github.com/user-attachments/assets/2d613031-0821-4e1c-aa3f-64edb8bdb3e2)

sarkaaa replied:
I am also wondering if it's not better to let there static `<Text>` element as it follows the prop type of <Loading> component. 🤔

sarkaaa replied:
ok, we can keep card as an example. 👍

sarkaaa commented on packages/orbit-components/src/Itinerary/Itinerary.stories.tsx:
```diff
@@ -385,163 +641,131 @@ export const ItinerarySeparatorComponent = () => {
-  hidden: false,
-};
+  args: {
+    color: "paletteRedDarkHover",
```
If I looked correctly, we have defined 77 variants of palette colors that could be used. It would be better to have it as a select list, to make it clearer for the user. I have to just think where to put the enum with defined colors, as I guess we don't wanna have ~80 additional lines just for an enum in `stories.tsx` file. 😄

sarkaaa commented on packages/orbit-components/src/Itinerary/Itinerary.stories.tsx:
```diff
@@ -385,163 +641,131 @@ export const ItinerarySeparatorComponent = () => {
-      <ItinerarySegmentDetail duration="2h 30m" summary={<BadgeGroup />} content={content} />
-    </ItinerarySegment>
-  );
+export const InsideModal: Story = {
```
What do you mean by that? This story includes an example of a basic modal window which is available from the itinerary component. The difference from current SB is that the modal is opened by default and you can close the original modal with itinerary to be able to copy the code.

domihustinova replied:
I mean that there are two modals over each other 🤔 https://kiwicom-orbit-sarka-storybook-migration-b4.surge.sh/?path=/story/itinerary--inside-modal

<img width="1440" alt="Snímek obrazovky 2024-09-02 v 22 12 42" src="https://github.com/user-attachments/assets/4010966a-84a0-4dc0-b73a-e21dfecf0f93">

sarkaaa replied:
yes, but it's based on the old Inside Modal story. I can set `false` to open state for the `Hidden city info` if it makes more sense to you. :D 🤷‍♀️

domihustinova replied:
Yes, definitely :D I completely forgot that there is second modal on the click. It makes sense to have the modals opened as you set, but for the small one I think it make sense to keep it closed and open only on click.

sarkaaa replied:
as we discussed in PM, I put Itinerary out of the modal wrapper and kept the inside modal. Modal wrapper + Itinerary are in `Modal` stories now.

domihustinova commented on packages/orbit-components/src/Itinerary/Itinerary.stories.tsx:
```diff
@@ -385,163 +641,131 @@ export const ItinerarySeparatorComponent = () => {
+  args: {
+    color: "paletteRedDarkHover",
+    type: "solid",
+  },
```
Omg I was looking at the `Separator`, not `ItinerarySeparator` 😅 🙈

sarkaaa commented on packages/orbit-components/src/Itinerary/Itinerary.stories.tsx:
```diff
@@ -163,210 +137,492 @@ const content = [
+    spaceAfter: SPACINGS_AFTER.MEDIUM,
+    noElevation: false,
+    actionable: true,
+    banner: undefined,
```
`banner` prop is static in this example. I didn't add it as a dynamic control as it's React.ReactNode type and it would be problematic for the user to set ReactNode prop in controls. I can remove the `banner` prop at all, but the reason behind this is that users can notice that they can use this prop for an `ItinerarySegment` component as well.

sarkaaa replied:
but it's true that arg control `banner` can be out of the specified `args` at all. 👍

sarkaaa commented on packages/orbit-components/src/Modal/Modal.stories.tsx:
```diff
@@ -392,228 +159,456 @@ export const WithFixedFooter = () => {
-        preventOverlayClose={preventOverlayClose}
-      >
-        {header && (
+export const WithoutSection: Story = {
```
It seems there's not any **significant** difference, I'll remove one of these stories.

sarkaaa commented:
ok, so I updated most of the stories (except Loading for now), I create 2nd commit for Modal, which I'll squash once it will be finalized.

In the modal:
- I kept the `WithForm` story to visualize example how modal and form can be used.
- I added the `With Itinerary` story to visualize an example of modal + itinerary (this story was in Itinerary before but it fits more for `Modal`)
- I removed `ShortModal`, `Sizes` as these functionalities can be simulated in the `Playground` story
- I removed `WithoutSections` story as it's almost same as `RemovableSections` story

sarkaaa commented on packages/orbit-components/src/NotificationBadge/NotificationBadge.stories.tsx:
```diff
@@ -8,238 +9,97 @@ import NotificationBadge from ".";
-  parameters: {
-    info: "Check Orbit.Kiwi for more detailed design guidelines.",
+    return (
+      <NotificationBadge type="info" ariaLabel={ariaLabel} icon={Icon && <Icon />}>
```
should be fixed. 🤞

sarkaaa commented on packages/orbit-components/src/NotificationBadge/NotificationBadge.stories.tsx:
```diff
@@ -8,238 +9,97 @@ import NotificationBadge from ".";
-  parameters: {
-    info: "Check Orbit.Kiwi for more detailed design guidelines.",
+    return (
+      <NotificationBadge type={type} ariaLabel={ariaLabel} icon={Icon && <Icon />}>
```
should be fixed. 🤞

sarkaaa commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
+export const Default: Story = {
   parameters: {
-    info: "Some description about this type of ListChoice in general.",
+    info: "Minimal setup with title, uncontrolled, no selectable. Check Orbit.Kiwi for more detailed design guidelines.",
```
should be fixed. 🤞

sarkaaa commented on packages/orbit-components/src/Itinerary/consts.ts:
```diff
@@ -0,0 +1,85 @@
+export enum STATUS_TYPES {
```
I've created this file to store enums that were placed in `stories.tsx` file. The reason behind this is that `PALETTER_COLORS` contains so many lines.

sarkaaa commented on packages/orbit-components/src/Itinerary/consts.ts:
```diff
@@ -0,0 +1,85 @@
+  NEUTRAL = "neutral",
+}
+
+export enum PALETTE_COLORS {
```
maybe I'm wrong and there is some plan I am not aware of yet, but I guess there won't be many changes in color palette naming (?). if yes, wouldn't be there same process as with the tokens and it can be potentially updated with codemod file? 🤔

sarkaaa commented:
Almost everything should be fixed, except the `text` prop for **Loading** - additional note for [this comment](https://github.com/kiwicom/orbit/pull/4447#discussion_r1741026206).

From my research, the previous SB included custom `loadingText` arg (type string), which was passed to `text` prop of Loading component. It seems that's the reason why it was accepted in older Storybook. You can check it [here](https://kiwicom.github.io/orbit/?path=/story/loading--playground) (`loadingText` arg in controls), the file is [here](https://github.com/kiwicom/orbit/blob/master/packages/orbit-components/src/Loading/Loading.stories.tsx).

I think we have these options:
1. Create custom arg `loadingText` as it is in the older version of Storybook.
2. Keep the static `<Text>` component in the code as it's the current PR's version (and have `text` control excluded from controls).
3. Keep `text` in controls as a string, but it seems that `"` would be needed.

I like the most the second option (as it's now) because we don't mix `loadingText` custom arg there, which can be confusing for the user. I don't like the third option.

What's your opinion? @domihustinova

domihustinova commented:
Regarding the `Loading`, have you tried restricting the type of `text` prop inside `argTypes` to be type `text`? I think it might help with the `"` 🤔

domihustinova commented on packages/orbit-components/src/NotificationBadge/NotificationBadge.stories.tsx:
```diff
@@ -8,238 +9,78 @@ import NotificationBadge from ".";
+export const Rtl: Story = {
+  render: () => (
+    <RenderInRtl>
+      <NotificationBadge type="info">10</NotificationBadge>
```
Why are props here static and not spread here as args? And same for Default story actually.

sarkaaa replied:
RTL:
We can include args there, I wanted to keep the basic minimal setup for RTL version. If we would like to go with `args` spreading in the code, I need to do an "additional" setup, like the args definition
```
args: {
  icon: undefined,
},
```

and defining `children` in render and child component (as it's in Playground story right now).
```
render: ({ children, ...args }) => (
    <RenderInRtl>
      <NotificationBadge {...args}>{children}</NotificationBadge>
      ...
```

from my point of view, I think that a minimal setup could be enough just for RTL demonstration and have to code as simple as possible.


**default story:**
all of the props (args) are not mandatory, I kept `children` to demonstrate that the NotificationBadge doesn't have to include _only_ icons.

domihustinova replied:
> If we would like to go with args spreading in the code, I need to do an "additional" setup

But why? If this is the default, then we can just use this default, no?

```
args: {
    children: "10",
    icon: "null",
    type: TYPE_OPTIONS.INFO,
    ariaLabel: "additional information for screen readers",
  },
```

And if the `icon` is an issue, it can be set to `undefined` on meta, no?

I am starting to get lost when do we want to use args and we are just hardcoding :D

domihustinova commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
-
-Rtl.story = {
-  name: "RTL",
+export const Rtl: Story = {
```
Why are all the args here hardcoded and not used from args?

domihustinova commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
-    />
-  );
+  args: {
+    selected: undefined,
```
If default is false, this is not necessary here, no? It should work even without it.

domihustinova commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
+    disabled: false,
+    tabIndex: 0,
+    onClick: action("onClick"),
+    action: null,
```
You set `action` here as null, but then in `WithAction` story you set it as undefined and you also exclude it in every story. Can't we just not set it at all?

sarkaaa replied:
GP, let's move it out.

domihustinova commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
+    title: "Choice Title",
+    description: "Further description",
+    icon: "Accommodation",
+    selectable: true,
```
I am trying to figure out how to not have so many `undefined` in the file. Maybe `selectable` could be set to false here? You are setting it to undefined/false in multiple places so maybe let's just set it to `true` when relevant?

`selected` is also excluded in a lot of stories, maybe it can be not set on meta but only in Playground? Maybe there could be more optimisation being made with other args as well.

sarkaaa replied:
> selected is also excluded in a lot of stories, maybe it can be not set on meta but only in Playground? Maybe there could be more optimisation being made with other args as well.

I thought we agreed at the start that we would define args in meta and then exclude for each of the stories where it's not needed.

domihustinova replied:
But that makes sense only when you have a lot of repeating args and it really makes sense to set it on meta so you are not repeating yourself. But if you set something on meta and then you are turning it off in every story, that doesn't really makes sense, no? :D

domihustinova commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,185 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
-Controlled.args = {
-  icon: "Accommodation",
-};
+export const Playground: Story = {
```
Why you didn't include `role` prop here?

domihustinova commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,113 @@
 
-InlineLoader.story = {
-  name: "Inline loader",
+export const Default: Story = {
```
Still visible 😅

domihustinova commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,152 @@
+    text: "Please wait, content is loading...",
+    customSize: 50,
+    type: TYPE_OPTIONS.PAGE_LOADER,
+    id: "loader-id",
```
I am thinking if it makes sense to even set all these args here on meta. We don't really need `id` anywhere except Playground, and in most stories you are setting them to `undefined` anyway. Maybe let's just set the default one from the component, so `loading` and `type`? WDYT?

domihustinova commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,152 @@
+    cardDescription: "Card description",
+    text: undefined,
+    customSize: undefined,
+  },
```
Are all these args necessary? Do we need to have Card props in controls? I would just keep here the control for `loading` and hardcode the rest, WDYT?

domihustinova commented on packages/orbit-components/src/Loading/Loading.stories.tsx:
```diff
@@ -1,123 +1,152 @@
+  args: {
+    loading: true,
+    type: TYPE_OPTIONS.INLINE_LOADER,
+    text: "Please wait, content of the page is loading...",
```
Not sure how you will end up arranging the props based on my suggestions, but just pointing out that here are almost the same args as currently set on meta. But if you end up removing them from meta, then it's OK if they are not repeating the default values.

sarkaaa replied:
now `text` will be included in `Default` story and I will re-use it in Playground using spreading.

sarkaaa commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,173 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
+  args: {
+    ...MultipleChoices.args,
+    selectable: false,
+    selected: false,
```
yes, it is, but I want to set `false` value for `selectable` as a default value for this specific story (to show the user that the Plus icon can be used in list choice). `selected` can be moved out, true, I didn't notice that. 👍

domihustinova replied:
Yes, yes, selectable is `true` in `MultipleChoices.args`, I was referring only to `selected` 👍

sarkaaa commented on packages/orbit-components/src/ListChoice/ListChoice.stories.tsx:
```diff
@@ -7,239 +9,173 @@ import RenderInRtl from "../utils/rtl/RenderInRtl";
+export const WithAction: Story = {
+  render: ({ icon, ...args }) => {
+    const Icon = typeof icon === "string" && getIcon(icon);
+    const IconButton = typeof icon === "string" && getIcon("Plus");
```
yes, sure. :D

sarkaaa commented on packages/orbit-components/src/Modal/Modal.stories.tsx:
```diff
@@ -558,14 +519,46 @@ export const Playground: PlagroundStory = {
+      },
     },
+    illustration: {
+      options: NAMES,
```
yes, why not. I did the same approach for icon prop in some other component. 👍

sarkaaa commented on packages/orbit-components/src/Modal/Modal.stories.tsx:
```diff
@@ -90,156 +70,6 @@ function useModal() {
-};
-
 export const RemovableSections: Story = {
   render: ({ showSection, children, lockScrolling }) => {
```
There was a general mess with using `lockScrolling`, so I changed usage at multiple places. :D
